plugins/UsersManager/API.php

```diff
@@ -795,26 +798,35 @@ public function regenerateTokenAuth($userLogin)
      * Updates a user in the database.
      * Only login and password are required (case when we update the password).
      *
-     * If the password changes and the user has an old token_auth (legacy MD5 format) associated,
-     * the token will be regenerated. This could break a user's API calls.
+     * If password or email changes, it is required to also specify the password of the current user needs to be specified
+     * to confirm this change.
      *
      * @see addUser() for all the parameters
      */
     public function updateUser($userLogin, $password = false, $email = false, $alias = false,
-                               $_isPasswordHashed = false)
+                               $_isPasswordHashed = false, $passwordConfirmation = false)
     {
+        $requirePasswordConfirmation = self::$UPDATE_USER_REQUIRE_PASSWORD_CONFIRMATION;
```

Comment:
Couldn't find a better way to make `updateUser` verify the password but sometimes not. Eg passwordResetter wouldn't know the current password of the user as it is stored hashed... 

things like only checking it when `Request::isRootRequestApiRequest()` would not work (cause some plugins call it through API) and neither `$requirePasswordConfirmation = Request::isCurrentApiRequestTheRootApiRequest()` cause eg because people could avoid it by using bulk requests etc.

Reply:
Could you just use the `UserUpdater` that you introduced? For PasswordResetter I assume it would be ok, since it doesn't use `Request::processRequest()`.

Reply:
Oh I see below that class just issues an API request. Could that class instead just use the model & set the password directly? I assume all it would need to do is hash the password and call Model::updateUser().

Reply:
I thought about it as well and then figured it may be beneficial to still use the API just to have all the permission checks running again, the code is covered by tests, deletes tracker cache after updating user etc. Otherwise might be too easy to regress something